### PR TITLE
JSAEM2-17 feat: add PersistStorage to load and store transactions

### PR DESCRIPTION
**Pull request by MichaelCTH**

### CODE REVIEW

File: src/TransCreator/index.jsx [hunk 1]
```diff
@@ -15,30 +15,32 @@ import PageHeader from './PageHeader';
...
-
+import { addSingleTransaction } from '../Stats/actionCreator';
+import { saveTransactionList } from '../PersistStorage';
 
 const TransCreator = ({ navigation }) => {
   const [transAmount, setTransAmount] = useState(null);
   const [transType, setTransType] = useState('Expense');
   const [transDate, setTransDate] = useState(moment().unix());
   const [newTransInsertionSuccess, setNewTransInsertionSuccess] = useState(false);
+
```
Comment (KatyTao) [inline src/TransCreator/index.jsx]: Please delete this blank space

**Comment by KatyTao** [comment]
Looks all good

File: src/Stats/actionCreator.jsx [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
     },
   };
 }
+
+export function loadTransaction(bulkTrans) {
+  return {
+    type: actionType.LOAD_LIST_TRANS,
+    data: bulkTrans,
+  };
+}
```
Comment (ikarasz) [inline src/Stats/actionCreator.jsx]: do we still need these changes?

File: src/Stats/reducer.jsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+    case actionType.ADD_SINGLE_TRANS:
       return {
         ...state,
-        transactions: [...state.transactions, { ...action.data }],
+        transactionList: [...state.transactionList, { ...action.data }],
+      };
+    case actionType.LOAD_LIST_TRANS:
+      return {
+        ...state,
+        transactionList: [...action.data],
```
Comment (ikarasz) [inline src/Stats/reducer.jsx]: do we still need these changes?

File: src/TransCreator/index.jsx [hunk 4]
```diff
@@ -15,8 +15,7 @@ import PageHeader from './PageHeader';
 import PageFooter from './PageFooter';
 import AmountInput from './AmountInput';
 import DateSelector from './DateSelector';
-import { addNewTransaction } from '../Stats/actionCreator';
-
+import { addSingleTransaction } from '../Stats/actionCreator';
```
Comment (ikarasz) [inline src/TransCreator/index.jsx]: do we need these changes?

File: src/TransCreator/index.jsx [hunk 5]
```diff
@@ -37,8 +36,7 @@ const TransCreator = ({ navigation }) => {
       Alert.alert('Please enter the amount');
       return;
     }
-
-    dispatch(addNewTransaction(transType, transDate, transAmount));
+    dispatch(addSingleTransaction(transType, transDate, transAmount));
```
Comment (ikarasz) [inline src/TransCreator/index.jsx]: do we need these changes?

File: src/reduxStore.jsx [hunk 6]
```diff
@@ -1,8 +1,20 @@
...
-const mainReducer = combineReducers({
-  transactions,
+const persistConfig = {
+  key: 'root',
+  storage,
+  stateReconciler: autoMergeLevel2,
+};
+
+const rootReducer = combineReducers({
+  transactionReducer,
```
Comment (ikarasz) [inline src/reduxStore.jsx]: Isn't it break our app if we rename the state?

File: src/Stats/actionCreator.jsx (hunk 2 above)
Comment (MichaelCTH) [reply src/Stats/actionCreator.jsx]: fixed

File: src/Stats/reducer.jsx (hunk 3 above)
Comment (MichaelCTH) [reply src/Stats/reducer.jsx]: We decided to use `transactionList ` instead to avoid the confusion with store name `transaction`

File: src/TransCreator/index.jsx (hunk 4 above)
Comment (MichaelCTH) [reply src/TransCreator/index.jsx]: fixed

File: src/reduxStore.jsx (hunk 6 above)
Comment (MichaelCTH) [reply src/reduxStore.jsx]: fixed

File: src/TransCreator/index.jsx (hunk 5 above)
Comment (MichaelCTH) [reply src/TransCreator/index.jsx]: fixed

File: src/reduxStore.jsx [hunk 7]
```diff
@@ -1,8 +1,20 @@
...
-const mainReducer = combineReducers({
-  transactions,
+const persistConfig = {
+  key: 'root',
+  storage,
+  stateReconciler: autoMergeLevel2,
+};
+
+const rootReducer = combineReducers({
+  transaction: transactionReducer,
```
Comment (ikarasz) [inline src/reduxStore.jsx]: it was transactions, do not change it, if it is not related to this story

File: src/Stats/reducer.jsx [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+  transactionList: [],
 };
 
 export default (state = initialState, action) => {
   switch (action.type) {
     case actionType.ADD_TRANS:
       return {
         ...state,
-        transactions: [...state.transactions, { ...action.data }],
+        transactionList: [...state.transactionList, { ...action.data }],
```
Comment (ikarasz) [inline src/Stats/reducer.jsx]: do not add unrelated change, if the team wants to rename it please create another branch and pr

File: src/reduxStore.jsx (hunk 7 above)
Comment (MichaelCTH) [reply src/reduxStore.jsx]: fixed

File: src/Stats/reducer.jsx (hunk 8 above)
Comment (MichaelCTH) [reply src/Stats/reducer.jsx]: fixed
